### PR TITLE
feat: split package fetcher

### PR DESCRIPTION
**Pull request by d3lm**
This PR splits the tarball fetcher into more specific types including
- `localTarball` for any locally hosted tarballs
- `remoteTarball` for packages hosted via remote tarballs, e.g. coming from the npm registry
- `gitHostedTarball` for git hosted packages

### CODE REVIEW

**Review by zkochan — APPROVED** [review]
lgtm

**Comment by d3lm** [comment]
Closed https://github.com/pnpm/pnpm/security/code-scanning/15 because it's only used in a test.

File: packages/node.fetcher/src/index.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@ import {
   RetryTimeoutOptions,
 } from '@pnpm/fetching-types'
 import { FilesIndex } from '@pnpm/fetcher-base'
+import { getFetcher } from '@pnpm/pick-fetcher'
```
Comment (zkochan) [inline packages/node.fetcher/src/index.ts]: I guess name the function pickFetcher

File: packages/pick-fetcher/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# @pnpm/pick-fetcher
+
+> Pick a tarball fetcher by type
```
Comment (zkochan) [inline packages/pick-fetcher/README.md]: Not a tarball fetcher but a package fetcher

File: packages/node.fetcher/src/index.ts (hunk 1 above)
Comment (d3lm) [reply packages/node.fetcher/src/index.ts]: 👍

**Review by zkochan — APPROVED** [review]

**Pull request merged**